Dev setup updates

QLLZ opened the pull request:
updated functions to pull sex/age-specific data

added reports for dropped observations in the pipeline.

QLLZ committed:
adding diagnostic report for dropped data in the mapping pipeline

QLLZ committed:
added functions to pull sex/age-specific data

QLLZ committed:
Update pull_data_helpers.R

updated functions to pull age/sex-specific data

QLLZ committed:
Update pull_data_helpers.R

converted to original file

QLLZ committed:
Update pull_data_helpers.R

QLLZ committed:
updating age specific analysis

QLLZ committed:
Update age_data_functions.R

QLLZ committed:
Update data_by_age_report.Rmd

eclee25 commented on Analysis/R/age_specific_analysis/age_data_functions.R:
```diff
@@ -0,0 +1,377 @@
+    raw == "2 1/2," ~ 2.5,
+    
+    # months (convert to years)
+    raw == "10 mo" ~ 10/12,
```
Would it make sense to do this more generically with a regular expression?

Remove all whitespaces for fields with letters, convert all letters to lower case, then separate string with numbers and text if they follow the "[0-9]m" pattern, and divide the numeric value by 12?

I think thiw would take care of all of the situations except 1 year & 8 months

eclee25 commented on Analysis/R/age_specific_analysis/age_data_functions.R:
```diff
@@ -0,0 +1,377 @@
+    
+    raw == "1 year & 8 months" ~ 1 + 8/12,
+    
+    str_detect(raw, regex("(jan|feb|mar|apr|may|jun|jul|aug|sep|sept|oct|nov|dec)", ignore_case = TRUE)) ~ NA_real_,
```
What is this for?

QLLZ replied:
there's a case where the month was part of the age column (like: 12-nov), so I was just removing the nov assuming it may mean the birth month.

eclee25 commented on Analysis/R/age_specific_analysis/age_data_functions.R:
```diff
@@ -0,0 +1,377 @@
+    
+    raw == "1 year & 8 months" ~ 1 + 8/12,
+    
+    str_detect(raw, regex("(jan|feb|mar|apr|may|jun|jul|aug|sep|sept|oct|nov|dec)", ignore_case = TRUE)) ~ NA_real_,
```
What is this for?

eclee25 commented on Analysis/R/age_specific_analysis/age_data_functions.R:
```diff
@@ -0,0 +1,377 @@
+    ) %>% 
+    gt() %>% 
+    cols_label(
+      minTL = 'Earlest TL',
```
typo: Earliest

eclee25 commented on Analysis/R/age_specific_analysis/age_data_functions.R:
```diff
@@ -0,0 +1,377 @@
+  out <- case_when(
+    raw %in% c("", "No data", "no data", "NA", "N/A", "s", ";") ~ NA_real_,
+    
+    raw == "5O" ~ 50,
```
Why are these weird characters here? Is this an issue with unicode?

QLLZ replied:
most of these strange characters are from OC 21124.

QLLZ committed:
Update age_data_functions.R

fixing some issues

QLLZ committed:
Update age_data_functions.R

QLLZ committed:
Update dropped_data_diagnostics.Rmd

QLLZ committed:
Update pull_data_helpers.R

updated sql data to extract some columns directly from the observations table not from the data column since for some observations, data column doesn't include all variables.

QLLZ committed:
Update dropped_data_diagnostics.Rmd

QLLZ committed:
Update dropped_data_diagnostics.Rmd
